src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -562,10 +585,26 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
                                     .commit(barrier.epoch)
                                     .await?;
 
-                                yield Message::Barrier(barrier);
-
-                                if self.backfill_finished(&backfill_stage.states).await? {
-                                    break 'backfill_loop;
+                                if self.should_report_finished(&backfill_stage.states) {
+                                    // TODO: use a specialized progress for source
+                                    // Currently, `CreateMviewProgress` is designed for MV backfill, and rw_ddl_progress calculates
+                                    // progress based on the number of consumed rows and an estimated total number of rows from hummock.
+                                    // For now, we just rely on the same code path, and for source backfill, the progress will always be 99.99%.
+                                    tracing::info!("progress finish");
+                                    let epoch = barrier.epoch;
+                                    self.progress.finish(epoch, 114514);
```

Comment:
The 2nd argument of `progress.finish()` is `current_consumed_rows`, which I think it's easy to count and report an accurate number. Why not doing it?

Reply:
You are right. Not did it just because no `total_consumed_rows` and thought it meaningless

Reply:
Following the https://github.com/risingwavelabs/risingwave/pull/18112#discussion_r1741571151 will make it more meaningful.